pkg/api/unversioned/group_version.go

```diff
@@ -198,3 +198,46 @@ func (gv *GroupVersion) UnmarshalJSON(value []byte) error {
 func (gv *GroupVersion) UnmarshalText(value []byte) error {
 	return gv.unmarshal(value)
 }
+
+// ToAPIVersionAndKind is a convenience method for satisfying runtime.Object on types that
+// do not use TypeMeta.
+func (gvk *GroupVersionKind) ToAPIVersionAndKind() (string, string) {
+	if gvk == nil {
+		return "", ""
+	}
+	return gvk.GroupVersion().String(), gvk.Kind
+}
+
+// FromAPIVersionAndKind returns a GVK representing the provided fields for types that
+// do not use TypeMeta.
+func FromAPIVersionAndKind(apiVersion, kind string) *GroupVersionKind {
```

Comment:
It's adapting internal old code.  I'll add a TODO that this should go away.

On Fri, Dec 11, 2015 at 8:11 AM, David Eads <notifications@github.com>
wrote:

> In pkg/api/unversioned/group_version.go
> <https://github.com/kubernetes/kubernetes/pull/18473#discussion_r47353057>
> :
>
> > @@ -198,3 +198,46 @@ func (gv *GroupVersion) UnmarshalJSON(value []byte) error {
> >  func (gv *GroupVersion) UnmarshalText(value []byte) error {
> >  	return gv.unmarshal(value)
> >  }
> > +
> > +// ToAPIVersionAndKind is a convenience method for satisfying runtime.Object on types that
> > +// do not use TypeMeta.
> > +func (gvk *GroupVersionKind) ToAPIVersionAndKind() (string, string) {
> > +	if gvk == nil {
> > +		return "", ""
> > +	}
> > +	return gvk.GroupVersion().String(), gvk.Kind
> > +}
> > +
> > +// FromAPIVersionAndKind returns a GVK representing the provided fields for types that
> > +// do not use TypeMeta.
> > +func FromAPIVersionAndKind(apiVersion, kind string) *GroupVersionKind {
>
> I'm less than thrilled with this. I was hoping to avoid having enough
> string representations of apiVersion and kind to ever want this method.
> Can this be made local to its point of use? Or maybe take a TypeMeta?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18473/files#r47353057>.
>